app/api/services/ocr/specs/OcrManager.spec.ts

```diff
@@ -0,0 +1,374 @@
+
+      await mocks.taskManagerMock.trigger(mockedMessageFromRedis);
+
+      const records = await OcrModel.get({});
+      expect(records).toHaveLength(3);
+      expect(filesApi.fileFromReadStream).not.toHaveBeenCalled();
+      expect(processDocumentApi.processDocument).not.toHaveBeenCalled();
+    });
+
+    it('message is not a success, and record error in db', async () => {
+      mocks.jestMocks['date.now'].mockReturnValue(1002);
+      await OcrModel.delete({ sourceFile: fixturesFactory.id('sourceFile') });
+
+      const [sourceFile] = await files.get({ _id: fixturesFactory.id('sourceFile') });
+      await OcrManager.addToQueue(sourceFile);
+      await mocks.taskManagerMock.trigger({
+        ...mockedMessageFromRedis,
+        success: false,
+        error: 'some error message',
+      });
+
+      const matchingRecords = await OcrModel.get({
+        sourceFile: fixturesFactory.id('sourceFile'),
+      });
+      expect(matchingRecords).toHaveLength(1);
+      const [record] = matchingRecords;
+      expect(record).toMatchObject({
+        status: 'cannotProcess',
+        sourceFile: fixturesFactory.id('sourceFile'),
+        language: 'eng',
+        lastUpdated: 1002,
+      });
+      expect(sockets.emitToTenant).toHaveBeenCalledWith(
+        tenantName,
+        'ocr:error',
+        sourceFile._id.toHexString()
+      );
+    });
+
+    it('there is an unexpected error and log', async () => {
+      const error = new Error('some error');
+      jest.spyOn(files, 'get').mockReturnValueOnce(Promise.reject(error));
+      await mocks.taskManagerMock.trigger(mockedMessageFromRedis);
+      expect(handleError.handleError).toHaveBeenCalledWith(error);
+    });
+  });
+
+  describe('on cleanup', () => {
+    it('when source is deleted, should modify record source to null', async () => {
+      const filesToCleanup = await files.get({
+        $or: [
+          { _id: fixturesFactory.id('sourceToDelete') },
+          { _id: fixturesFactory.id('sourceToDelete2') },
+        ],
+      });
+      let records = await OcrModel.get({ sourceFile: { $in: filesToCleanup.map(f => f._id) } });
+      await OcrManager.cleanupRecordsOfFiles(filesToCleanup.map(f => f._id));
+      records = await OcrModel.get({ _id: { $in: records.map(r => r._id) } });
+      expect(records[0].sourceFile).toBeNull();
+      expect(records[1].sourceFile).toBeNull();
+    });
+
+    it('when result is deleted, should delete record', async () => {
+      const filesToCleanup = await files.get({
+        $or: [
+          { _id: fixturesFactory.id('resultToDelete') },
+          { _id: fixturesFactory.id('resultToDelete2') },
+        ],
+      });
+      let records = await OcrModel.get({ sourceFile: { $in: filesToCleanup.map(f => f._id) } });
+      await OcrManager.cleanupRecordsOfFiles(filesToCleanup.map(f => f._id));
+      records = await OcrModel.get({ _id: { $in: records.map(r => r._id) } });
+      expect(records).toHaveLength(0);
+    });
+  });
+});
```

Comment:
This entire file has the wrong naming conventions on describes and its.

Every `describe` is a scenario or conditional: "on error", "when there is no tenant", "if promise rejected".  It should never start with a `should`.  Every `it`, on the other hand, MUST start with a `should`.

On rare occasions, an `it` also involves a single conditional. In order to avoid a new `describe` with a single `it`, we put the condition after the `should`.  So: "on cleanup" -> "should modify record source to null when source is deleted".

But always start with a should.

Reply:
Will take a look